### PR TITLE
build(ISD-281): Remove .flake8, change pyproject.toml and add docstring

### PR DESCRIPTION
**Pull request by amandahla**
Handle some reviews from:
https://github.com/canonical/content-cache-k8s-operator/pull/5
https://github.com/canonical/content-cache-k8s-operator/pull/7

* Remove .flake8
* Add `per-file-ignores = ["tests/*:D100,D101,D102,D103,D104,D205,D212,D415"] to pyproject.toml
* Add docstring to nginx_integrator_app

### CODE REVIEW

**Review by nrobinaubertin — APPROVED** [review]
LGTM

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: pyproject.toml [hunk 1]
```diff
@@ -39,7 +39,7 @@ select = ["E", "W", "F", "C", "N", "R", "D", "H"]
 # Ignore D107 Missing docstring in __init__
 ignore = ["W503", "E501", "D107"]
 # D100, D101, D102, D103: Ignore missing docstrings in tests
```
Comment (merkata) [inline pyproject.toml:41]: Can you please add description of the ignored tests as a comment, I see D104, D205, D212 and D415 are missing such.

Comment (amandahla) [reply pyproject.toml]: Sure.

File: pyproject.toml [hunk 2]
```diff
@@ -38,7 +38,10 @@ select = ["E", "W", "F", "C", "N", "R", "D", "H"]
 # Ignore W503, E501 because using black creates errors with this
 # Ignore D107 Missing docstring in __init__
 ignore = ["W503", "E501", "D107"]
-# D100, D101, D102, D103: Ignore missing docstrings in tests
+# D100, D101, D102, D103, D104: Ignore missing docstrings in tests
+# D205: Ignore 1 blank line required between summary line and description in tests
```
Comment (arturo-seijas) [inline pyproject.toml:42]: Can't we just adapt out docstring to comply? Or do we not want to follow these?

Comment (amandahla) [reply pyproject.toml:42]: From the [Contributing](https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#test-structure), this is the docstring to be used in tests
```
    """
    arrange: given 2 numbers
    act: when they are added
    assert: the result must be the sum of the 2 numbers. If a description spans
        multiple lines, lines after the first should be indented by the default
        spacing of the file to indicate that the description continues.
    """
```
Flake8 is not so happy about it.

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]

**Pull request merged**